supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesInitializationActor.scala

```diff
@@ -94,7 +96,9 @@ class JesInitializationActor(override val workflowDescriptor: BackendWorkflowDes
       val upload = () => Future(path.writeAsJson(content))
 
       workflowLogger.info(s"Creating authentication file for workflow ${workflowDescriptor.id} at \n ${path.toString}")
-      Retry.withRetry(upload, isFatal = isFatalJesException, isTransient = isTransientJesException)(context.system) map { _ => () }
+      Retry.withRetry(upload, isFatal = isFatalJesException, isTransient = isTransientJesException)(context.system) map { _ => () } recover {
+        case failure => throw new IOException("Failed to upload authentication file", failure)
```

Comment:
I always get confused. Is this equivalent to 
```
  recoverWith {
    case failure => Future.failure(new IOException("Failed to upload authentication file", failure))
```
Except that recoverWith doesn't need to do the whole throw/catch dance?